src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -782,7 +782,7 @@ static int do_map(int argc, const char *argv[], Config *cfg)
     }
   } else {
     r = sscanf(cfg->devpath.c_str(), "/dev/nbd%d", &index);
-    if (r < 0) {
+    if (r <= 0) {
```

Comment:
This is failing the ```rbd-nbd.sh``` test [1] -- if ```r == 0```, you should set the ```r = EINVAL``` or something similar to ensure an error is returned to the shell.

[1] http://qa-proxy.ceph.com/teuthology/jdillaman-2019-04-12_14:00:27-rbd-wip-jd-testing-distro-basic-smithi/3838738/teuthology.log